File: app/src/main/java/com/kksionek/gdzietentramwaj/view/MenuItemRefreshCtrl.java
package com.kksionek.gdzietentramwaj.view;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;

import com.kksionek.gdzietentramwaj.R;

public class MenuItemRefreshCtrl {
    private final Context mCtx;
    private final MenuItem mMenuItem;
    private Animation mRotationAnimation = null;
    private ImageView mRefreshImage = null;

    public MenuItemRefreshCtrl(@NonNull Context ctx, @NonNull MenuItem menuItem) {
        mCtx = ctx;
        mMenuItem = menuItem;
    }

    public void startAnimation() {
        if (mRefreshImage == null) {
            LayoutInflater inflater = (LayoutInflater) mCtx.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            mRefreshImage = (ImageView) inflater.inflate(R.layout.refresh_action_view, null);
        }
        if (mRotationAnimation == null)
            mRotationAnimation = AnimationUtils.loadAnimation(mCtx, R.anim.anim_rotate);
        if (mMenuItem.getActionView() == null) {
            mRefreshImage.startAnimation(mRotationAnimation);
            mMenuItem.setActionView(mRefreshImage);
            mMenuItem.setEnabled(false);
        }
    }

    public void endAnimation() {
        if (mMenuItem.getActionView() != null) {
            mMenuItem.getActionView().clearAnimation();
            mMenuItem.setActionView(null);
        }
        mMenuItem.setEnabled(true);
    }
}

File: app/src/main/java/com/kksionek/gdzietentramwaj/map/view/MenuItemRefreshCtrl.kt
package com.kksionek.gdzietentramwaj.map.view

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.ImageView

import com.kksionek.gdzietentramwaj.R

@SuppressLint("InflateParams")
class MenuItemRefreshCtrl(
    private val context: Context,
    private val menuItem: MenuItem
) {
    private val rotationAnimation: Animation by lazy {
        AnimationUtils.loadAnimation(context, R.anim.anim_rotate)
    }
    private val refreshImage: ImageView by lazy {
        LayoutInflater.from(context).inflate(R.layout.view_refresh_action, null) as ImageView
    }

    fun startAnimation() {
        if (menuItem.actionView == null) {
            refreshImage.startAnimation(rotationAnimation)
            menuItem.actionView = refreshImage
        }
        menuItem.isEnabled = false
    }

    fun endAnimation() {
        if (menuItem.actionView != null) {
            menuItem.actionView.clearAnimation()
            menuItem.actionView = null
        }
        menuItem.isEnabled = true
    }
}
